lxd/instance/drivers/driver_qemu.go

```diff
@@ -2185,7 +2185,16 @@ func (d *qemu) deviceAttachPath(deviceName string, configCopy map[string]string,
 		return fmt.Errorf("Failed to connect to QMP monitor: %w", err)
 	}
 
-	addr, err := net.ResolveUnixAddr("unix", virtiofsdSockPath)
+	virtiofsdSocketDir, err := os.Open(filepath.Dir(virtiofsdSockPath))
```

Comment:
Can we use O_PATH approach here as well?

Reply:
yes, it should be

Reply:
ran some manual tests to make sure and indeed worked as expected